boards/DIM/Inc/App/App_ErrorTable.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <stdint.h>
+
+// clang-format off
+enum ErrorId
+{
+    BMS_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1HZ = 0,
+    BMS_CRITICAL_ERROR_DUMMY_CRITICAL_ERROR = 1,
+
+    DCM_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1HZ = 0,
+    DCM_CRITICAL_ERROR_DUMMY_CRITICAL_ERROR = 1,
+
+    FSM_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1HZ = 0,
+    FSM_CRITICAL_ERROR_DUMMY_CRITICAL_ERROR = 1,
+
+    PDM_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1HZ = 0,
+    PDM_CRITICAL_ERROR_DUMMY_CRITICAL_ERROR = 1,
+
+    DIM_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1HZ = 0,
+    DIM_CRITICAL_ERROR_DUMMY_CRITICAL_ERROR = 1,
+
+    NUM_ERRORS = 10, // Number of non-critical and critical errors combined
+                     // across ALL boards
+};
+
+struct Error
+{
+    // CAN message ID
+    uint32_t std_id;
+    // Each error ID is unique in the error table
+    enum ErrorId error_id;
+};
+
+struct ErrorTable* App_ErrorTable_Create(void);
+void App_ErrorTable_Destroy(struct ErrorTable* error_table);
+
+/**
+ * Check if there is any non-critical errors in the given error table, and if
+ * necessary, write the non-critical errors into the given error array
+ * @note The array must hold up to NUM_ERRORS of elements
+ * @param error_table The error table
+ * @param errors Any non-critical errors will be set in this error array
+ * @return The number of non-critical errors
+ */
+uint32_t App_ErrorTable_HasNonCriticalError(struct ErrorTable* error_table, struct Error errors[]);
+
+/**
+ * Check if there is any critical errors in the given error table, and if
+ * necessary, write the critical errors into the given error array
+ * @note The array must hold up to NUM_ERRORS of elements
+ * @param error_table The error table
+ * @param errors Any critical errors will be set in this error array
+ * @return The number of critical errors
+ */
+uint32_t App_ErrorTable_HasCriticalError(struct ErrorTable* error_table, struct Error errors[]);
+
+void App_ErrorTable_SetBmsStackWaterMarkAboveThresholdTask1Hz(struct ErrorTable* error_table);
```

Comment:
I'm confused why we're setting errors like this, but then define errors using the struct above? I originally proposed using the struct above so that we wouldn't need to have semantic understanding of what each error is.

Reply:
Did you mean `struct Error`? I made these setters becuase I thought it's what you meant by

>// Setters for each possible error message ... we would determine these by regex match on the message ID's from the dbc

I must have misinterpreted what you meant? Maybe `Io_ErrorTable_Update` is what you wanted me to write?